cli/src/semdep/parse_lockfile.py

```diff
@@ -45,38 +51,40 @@ def extract_npm_lockfile_hash(s: str) -> Dict[str, List[str]]:
     return {algorithm: [base64.b16encode(decode_base_64).decode("ascii").lower()]}
 
 
-def parse_Yarnlock_str(lockfile_text: str) -> Generator[FoundDependency, None, None]:
-    def extract_yarn_name(line: str) -> str:
-        """
-        "@babel/code-frame@^7.0.0", "@babel/code-frame@^7.10.4", "@babel/code-frame@^7.8.3":
-        should parse as @babel/code-frame
-
-        "@babel/types@^7.0.0", "@babel/types@^7.10.4", "@babel/types@^7.10.5", "@babel/types@^7.11.0", "@babel/types@^7.3.0", "@babel/types@^7.3.3", "@babel/types@^7.4.0", "@babel/types@^7.4.4", "@babel/types@^7.7.0", "@babel/types@^7.9.0":
-        should parse as @babel/types
-
-        lodash@4.17.18:
-        should parse as lodash
-        """
-        if '"' not in line:
-            return line.split("@")[0]
-        else:
-            first_quoted = "".join(line.split('"')[1:2])
-            parsed_name = "@".join(first_quoted.split("@")[:-1])
-            return parsed_name
+def parse_yarn(
+    lockfile_text: str, manifest_text: Optional[str]
+) -> Generator[FoundDependency, None, None]:
+    def version_sources(line: str) -> Tuple[str, List[Tuple[str, str]]]:
+        constraint_strs = line[:-1].split(",")
+        constraints = []
+        for c in constraint_strs:
+            c = c.strip().strip('"')
```

Comment:
Does .strip("\" ") work or is it possible there are spaces after "

Reply:
no that should work